### PR TITLE
DAOS-6924 test: Offline extend (More tests)

### PR DESCRIPTION
**Pull request by rpadma2**
Test-tag-hw-large: pr,hw,large offline_extend
Skip-unit-tests: true
Skip-func-test: true
Skip-func-hw-test-small: true
Skip-func-hw-test-medium: true

Summary:
   - Updated the offline extend test to use IOR/mdtest
   - Test with multiple pools
   - Test with and without checksum enabled.
   - Test offline extend with different object class
   - Perform aggregation during offline extend operation.

Signed-off-by: rpadma2 <ravindran.padmanabhan@intel.com>

### CODE REVIEW

File: src/tests/ftest/osa/osa_offline_extend.py [hunk 1]
```diff
@@ -22,83 +24,190 @@ def setUp(self):
...
         # Start the additional servers and extend the pool
         self.log.info("Extra Servers = %s", self.extra_servers)
         self.start_additional_servers(self.extra_servers)
         # Give sometime for the additional server to come up.
-        time.sleep(5)
+        for retry in range(0, 10):
+            scan_info = self.get_dmg_command().system_query()
+            if not check_system_query_status(scan_info):
+                if retry == 9:
+                    self.fail("One or more servers not in expected status")
```
Comment (wangdi1) [inline src/tests/ftest/osa/osa_offline_extend.py:87]: I thought you can break the for loop if check_system_query_status() return true?

**Review by wangdi1 — DISMISSED** [review]

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:87]: Yes, you are right... Will fix it...Thanks.

File: src/tests/ftest/osa/osa_offline_extend.py [hunk 2]
```diff
@@ -22,83 +24,190 @@ def setUp(self):
...
+        tmp = server_boot
+        self.log.info(tmp)
+        if oclass is None:
+            oclass = []
+            oclass.append(self.ior_cmd.dfs_oclass.value)
 
-        # Extend a ranks 4 and 5
-        rank = [4, 5]
+        self.log.info(oclass[0])
+        # tmpfile = self.ior_cmd.test_file.value
```
Comment (shimizukko) [inline src/tests/ftest/osa/osa_offline_extend.py:58]: Do we need this?

File: src/tests/ftest/osa/osa_offline_extend.py [hunk 3]
```diff
@@ -22,83 +24,190 @@ def setUp(self):
...
             data (bool) : whether pool has no data or to create
                           some data in pool. Defaults to False.
+            server_boot (bool) : Perform system stop/start on a rank.
+                                 Defaults to False.
+            oclass (list) : list of daos object class (eg: "RP_2G8")
         """
         # Create a pool
         pool = {}
-        pool_uuid = []
+        tmp = server_boot
```
Comment (shimizukko) [inline src/tests/ftest/osa/osa_offline_extend.py:51]: Not sure why you need to use this "tmp" variable. "server_boot" is more intuitive.

File: src/tests/ftest/osa/osa_offline_extend.py [hunk 4]
```diff
@@ -22,83 +24,190 @@ def setUp(self):
...
+        # tmpfile = self.ior_cmd.test_file.value
 
         for val in range(0, num_pool):
+            # Perform IOR write using the oclass list
+            if val < len(oclass):
+                index = val
+            else:
+                index = 0
+            # tstring = "{}".format(val)
+            # self.ior_cmd.test_file.value = tmpfile.join([tstring])
```
Comment (shimizukko) [inline src/tests/ftest/osa/osa_offline_extend.py:67]: Do we need these?

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 4 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:67]: Yes, they are not needed. Will fix it.

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 3 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:51]: hmm... Probably, I started thinking about something and left it... Yes, there is no need for tmp. Will fix it. Thanks.

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 2 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:58]: Will remove it. Thanks.

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:87]: Done.

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 2 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:58]: Done.

File: src/tests/ftest/osa/osa_offline_extend.py (hunk 4 above)
Comment (rpadma2) [reply src/tests/ftest/osa/osa_offline_extend.py:67]: Done.

**Review by shimizukko — APPROVED** [review]

**Review by wangdi1 — APPROVED** [review]

**Pull request merged**